core/src/main/kotlin/org/evomaster/core/output/TestSuiteOrganizer.kt

```diff
@@ -70,39 +73,84 @@ class NamingHelper {
         else return ""
     }
 
-    val namingCriteria =  mutableListOf(::criterion1_500, ::criterion2_hasPost, ::criterion3_sampling, ::criterion4_dbInit)
+    private var namingCriteria =  mutableListOf(::criterion1_500, ::criterion3_sampling)
+    private val availableCriteria = mutableListOf(::criterion1_500, ::criterion2_hasPost, ::criterion3_sampling, ::criterion4_dbInit)
 
 
     fun suggestName(individual: EvaluatedIndividual<*>): String{
         return namingCriteria.map { it(individual) }.joinToString("")
     }
 
+    fun getAvailableCriteria(): MutableList<KFunction1<EvaluatedIndividual<*>, String>> {
+        return availableCriteria
+    }
+
+    fun selectCriteria(selected: MutableList<KFunction1<EvaluatedIndividual<*>, String>>){
+        if (availableCriteria.containsAll(selected)){
+            namingCriteria = availableCriteria
+        }
+        else {
+            throw UnsupportedOperationException("The naming criteria chosen appear to not be supported at the moment.")
+        }
+    }
+
+    fun selectCriteriaByIndex(selected: MutableList<Int>){
```

Comment:
you are not modifying such input list, so why passing it as `MutableList` instead of a `List`?